pkg/prometheus/statefulset.go

```diff
@@ -431,6 +431,12 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 		}
 	}
 
+	if p.Spec.AllowOverlappingBlocks {
+		promArgs = append(promArgs, "-storage.tsdb.allow-overlapping-blocks")
+	} else {
```

Comment:
To maintain compatibility with old Prometheus versions (< 2.8), I'd prefer that we leave this field unset when the spec field is set to false.

Reply:
Good point!